bayes_opt/acquisition.py

```diff
@@ -0,0 +1,854 @@
+"""Acquisition functions for Bayesian Optimization."""
+import warnings
+import numpy as np
+from numpy.random import RandomState
+from scipy.optimize import minimize
+from scipy.stats import norm
+from scipy.special import softmax
+from .target_space import TargetSpace
+from .constraint import ConstraintModel
+from sklearn.gaussian_process import GaussianProcessRegressor
+from typing import Callable, List, Union, Tuple
+from copy import deepcopy
+from numbers import Number
+
+class ConstraintNotSupportedError(Exception):
+    """Raised when constrained optimization is not supported."""
+
+    pass
+
+
+class NoValidPointRegisteredError(Exception):
+    """Raised when an acquisition function depends on previous points but none are registered."""
+
+    pass
+
+
+class TargetSpaceEmptyError(Exception):
+    """Raised when the target space is empty."""
+
+    pass
+
+
+class AcquisitionFunction():
+    """Base class for acquisition functions.
+
+    Parameters
+    ----------
+    random_state : int, RandomState, default None
+        Set the random state for reproducibility.
+    """
+
+    def __init__(self, random_state=None):
+        if random_state is not None:
+            if isinstance(random_state, RandomState):
+                self.random_state = random_state
+            else:
+                self.random_state = RandomState(random_state)
+        else:
+            self.random_state = RandomState()
+        self.i = 0
+
+    def base_acq(self, *args, **kwargs):
+        """Provide access to the base acquisition function."""
+        raise NotImplementedError
```

Comment:
I think that `@abstractmethod` would be more appropriate here. `NotImlpementedError` to my understanding is intended to be used in situations where the functionality WILL eventually be implemented.

Reply:
done.